From e10935b4c7cbf5920bd7707ff61871bd25f2417c Mon Sep 17 00:00:00 2001
From: Christoph Reiter <reiter.christoph@gmail.com>
Date: Sat, 13 Mar 2021 11:26:39 +0100
Subject: [PATCH] templates: fix number formating for the js graph

In case the numbers go over 1000 we get thousand delimiters which aren't
valid javascript breaking the graph. Format the values manually instead.
---
 app/templates/module_graph_js.tpl | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/app/templates/module_graph_js.tpl b/app/templates/module_graph_js.tpl
index 9247ac4..1ec71c6 100644
--- a/app/templates/module_graph_js.tpl
+++ b/app/templates/module_graph_js.tpl
@@ -17,7 +17,7 @@
 
             data.addRows([
 {foreach from=$hourlyGraphData key=key item=value}
-                [new {$value.label}, {$value.rx}, {$value.tx}, {$value.total}],
+                [new {$value.label}, {$value.rx|string_format:"%f"}, {$value.tx|string_format:"%f"}, {$value.total|string_format:"%f"}],
 {/foreach}
             ]);
 
@@ -69,7 +69,7 @@
 
             data.addRows([
 {foreach from=$dailyGraphData key=key item=value}
-                [new {$value.label}, {$value.rx}, {$value.tx}, {$value.total}],
+                [new {$value.label}, {$value.rx|string_format:"%f"}, {$value.tx|string_format:"%f"}, {$value.total|string_format:"%f"}],
 {/foreach}
             ]);
             
@@ -116,7 +116,7 @@
 
             data.addRows([
 {foreach from=$monthlyGraphData key=key item=value}
-                [new {$value.label}, {$value.rx}, {$value.tx}, {$value.total}],
+                [new {$value.label}, {$value.rx|string_format:"%f"}, {$value.tx|string_format:"%f"}, {$value.total|string_format:"%f"}],
 {/foreach}
             ]);
 
-- 
2.30.2

